docs/dev/data_validation.rst

```diff
@@ -0,0 +1,688 @@
+.. _data_validation:
+
+================================================================================
+Data Validation Tests
+================================================================================
+
+We use a tool called `dbt <https://www.getdbt.com/>`_ to manage our data validation
+tests. dbt is a SQL-based data transformation tool that can be used to manage large data
+pipelines, but we're currently only using it for data validation.
+
+In dbt, every data validation test is a SQL query meant to select rows that fail the
+test. A successful test will return no results. The query can be parametrized so it can
+be reused across multiple tables. dbt includes a few `built-in data test definitions
+<https://docs.getdbt.com/docs/build/data-tests>`_, and the `dbt_expectations
+<https://github.com/metaplane/dbt-expectations>`_ package provides many more. We also
+define our own `custom data tests
+<https://docs.getdbt.com/best-practices/writing-custom-generic-tests>`_.
+
+.. note:: Why use dbt and SQL instead of Python?
+
+   Modern analytical query engines are extremely fast and memory efficient. By using dbt
+   with `DuckDB <https://duckdb.org/>`_ to query the PUDL `Parquet
+   <https://parquet.apache.org/>`_ outputs we're able to run thousands of validations
+   across hundreds of tables with billions of rows in a minute, instead of the 2-3 hours
+   it used to take our much less extensive validation tests to run. Plus we get to learn
+   SQL.
+
+--------------------------------------------------------------------------------
+Data validation guidelines
```

Comment:
We should come up with something for this section, but that's probably beyond the scope of this PR.